bson/bsoncodec/string_codec.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsoncodec
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.mongodb.org/mongo-driver/bson/bsonoptions"
+	"go.mongodb.org/mongo-driver/bson/bsonrw"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+)
+
+var defaultStringCodec = &StringCodec{true}
+
+// StringCodec is the Codec used for struct values.
+type StringCodec struct {
+	ObjectIDAsHex bool
+}
+
+var _ ValueCodec = &StringCodec{}
+
+// NewStringCodec returns a StringCodec with options opts.
+func NewStringCodec(opts ...*bsonoptions.StringCodecOptions) (*StringCodec, error) {
+	stringOpt := bsonoptions.MergeStringCodecOptions(opts...)
```

Comment:
nit: We could handle defaults in the `Merge*Options` functions themselves (e.g. https://github.com/mongodb/mongo-go-driver/blob/master/mongo/options/gridfsoptions.go#L40).

Reply:
done, added to bsonoptions.StringCodec()